bundles/org.eclipse.swt/Eclipse SWT Custom Widgets/common/org/eclipse/swt/custom/CTabFolderRenderer.java

```diff
@@ -1354,10 +1354,18 @@ void drawSelected(int itemIndex, GC gc, Rectangle bounds, int state ) {
 					}
 				}
 
-				//Highlight MUST be drawn before the outline so that outline can cover it in the right spots (start of swoop)
+				//Highlight colors MUST be drawn before the outline so that outline can cover it in the right spots (start of swoop)
 				//otherwise the curve looks jagged
 				drawHighlight(gc, bounds, state, rightEdge);
 
+				// draw highlight marker of selected tab
+				if (parent.simple) {
```

Comment:
Draw highlight marker only when thinkness is greater than 0:
if (parent.selectionHighlightBarThickness > 0 && parent.simple) {